### PR TITLE
feat: modularize client dbdump

### PR DESCRIPTION
**Pull request by m1sterc001guy**
Fixes: https://github.com/fedimint/fedimint/issues/3204

Also adds support for dumping gatewayd's database.

Moved `dump_database` to a common location for both the server modules and the client modules to implement. `dbtool` still just attaches the default modules (mint, ln, wallet). If more modules are needed it should be straightforward to fork and add the modules, with little code change.

### CODE REVIEW

File: fedimint-rocksdb/src/lib.rs [hunk 1]
```diff
@@ -31,10 +32,10 @@ impl RocksDb {
 }
 
 impl RocksDbReadOnly {
-    pub fn open_read_only(db_path: impl AsRef<Path>) -> Result<RocksDbReadOnly, rocksdb::Error> {
+    pub fn open_read_only(db_path: String) -> Result<RocksDbReadOnly, rocksdb::Error> {
```
Comment (elsirion) [inline fedimint-rocksdb/src/lib.rs:35]: Why not `PathBuf`? I'm also not a fan of hiding this change in this commit, could use some explanation in a commit message.

**Review by elsirion — APPROVED** [review]
Looks ok, not a big fan of the read only DB change, would like to know the reasoning.

File: fedimint-rocksdb/src/lib.rs (hunk 1 above)
Comment (m1sterc001guy) [reply fedimint-rocksdb/src/lib.rs:35]: Let me see if I can fix this before merging

Comment (m1sterc001guy) [reply fedimint-rocksdb/src/lib.rs]: Sorry I was being dumb, these changes aren't necessary. Reverted all changes in `fedimint-rocksdb`

**Comment by m1sterc001guy** [comment]
> Looks ok, not a big fan of the read only DB change, would like to know the reasoning.

DB change was not necessary and has been removed

**Review by elsirion — APPROVED** [review]

**Pull request merged**